# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/base/test/gtest/TestTextDirective.cpp
# Commit: 09616085d8ef
# Full Hash: 09616085d8ef6470309e1b3c565242d6a811cd5d
# Author: Jan-Niklas Jaeschke <jjaschke@mozilla.com>
# Date: 2025-07-22 09:32:06
# Description:
#   Bug 1978235 - Text Fragments: Ensure that start and end terms are always non-empty. r=mccr8,dom-core
#   
#   The patch for Bug 1976473 changed  the behavior of the algorithm to collect word boundary distances
#   to merge punctuation characters with the preceding word. However, this led to unexpected behavior if a
#   context term only consisted of punctuation, where this function would unexpectedly return an empty array.
# ==============================================================================

diff -r df6c5c98b663 -r 09616085d8ef dom/base/test/gtest/TestTextDirective.cpp
--- a/dom/base/test/gtest/TestTextDirective.cpp	Mon Jul 21 18:33:07 2025 +0000
+++ b/dom/base/test/gtest/TestTextDirective.cpp	Mon Jul 21 18:36:11 2025 +0000
@@ -15,13 +15,14 @@
   nsTArray<uint32_t> wordEndDistances =
       TextDirectiveUtil::ComputeWordBoundaryDistances<TextScanDirection::Right>(
           text);
-  EXPECT_EQ(wordEndDistances.Length(), 6u);
+  EXPECT_EQ(wordEndDistances.Length(), 7u);
   EXPECT_EQ(wordEndDistances[0], 5u);   // "Hello"
   EXPECT_EQ(wordEndDistances[1], 12u);  // "Hello, world"
   EXPECT_EQ(wordEndDistances[2], 18u);  // "Hello, world! This"
   EXPECT_EQ(wordEndDistances[3], 21u);  // "Hello, world! This is"
   EXPECT_EQ(wordEndDistances[4], 23u);  // "Hello, world! This is a"
   EXPECT_EQ(wordEndDistances[5], 28u);  // "Hello, world! This is a test"
+  EXPECT_EQ(wordEndDistances[6], 29u);  // "Hello, world! This is a test."
 }
 
 TEST(TestTextDirective, ComputeWordBoundaryDistancesRTL)
@@ -38,3 +39,23 @@
   EXPECT_EQ(wordBeginDistances[4], 22u);  // "world! This is a test."
   EXPECT_EQ(wordBeginDistances[5], 29u);  // "Hello, world! This is a test."
 }
+
+TEST(TestTextDirective, ComputeWordBoundaryDistancesPunctuationOnly)
+{
+  nsString text(u": , .");
+  nsTArray<uint32_t> wordEndDistances =
+      TextDirectiveUtil::ComputeWordBoundaryDistances<TextScanDirection::Right>(
+          text);
+  EXPECT_EQ(wordEndDistances.Length(), 1u);
+  EXPECT_EQ(wordEndDistances[0], 5u);
+}
+
+TEST(TestTextDirective, ComputeWordBoundaryDistancesWithEmptyString)
+{
+  nsString text(u"");
+  nsTArray<uint32_t> wordEndDistances =
+      TextDirectiveUtil::ComputeWordBoundaryDistances<TextScanDirection::Right>(
+          text);
+  EXPECT_EQ(wordEndDistances.Length(), 1u);
+  EXPECT_EQ(wordEndDistances[0], 0u);
+}